### ISSUE TITLE
Bootstrap fails on Vsphere IPI

### ISSUE THREAD

**Issue by lbrigman124** [openshift/okd#276]
**Describe the bug**
Single node Vsphere installation with IPI fails at bootstrap step unable to connect

**Version**
4.5.0-0.okd-2020-07-14-153706-ga

**How reproducible**
Ran twice with the same results

**Log bundle**
[log-bundle-20200721154136.tar.gz](https://github.com/openshift/okd/files/4961587/log-bundle-20200721154136.tar.gz)

**Comment by vrutkovs** [comment]
Master has requested its Ignition:
>I0721 22:00:55.645021       1 api.go:111] Pool master requested by address:"10.109.180.68:43706" User-Agent:"Ignition/2.3.0"

but log bundle didn't collect any data from it. Did it boot?
See [manual collection steps](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-manually-gathering-logs-with-SSH_installing-troubleshooting)

**Comment by beandj** [comment]
Same here ! 

[log-bundle-20200722150101.tar.gz](https://github.com/openshift/okd/files/4961699/log-bundle-20200722150101.tar.gz)

**Comment by lbrigman124** [comment]
Both the bootstrap and the master node show up in Vsphere as running.  I can connect to it.
The master only has one container and the logs are corrupt.

**Comment by vrutkovs** [comment]
>The master only has one container and the logs are corrupt.

That's probably a container with `machine-config-operator`, attempting to extract the daemon

**Comment by lbrigman124** [comment]
I'm seeing keepalived in the crio service journal.  Is there any offset detection for this?  We have multiple systems on this network that are running ipfailover pods from OKD 3.11 where at least one of the offsets is 0 which would make this a conflict.

**Comment by lbrigman124** [comment]
A small portion of output from:'journalctl -b -f -u kubelet.service -u crio.service'
Jul 22 18:43:47 lbgsm9-hh4f4-master-0 hyperkube[1620]: I0722 18:43:47.855050    1620 csi_plugin.go:938] Failed to contact API server when waiting for CSINode publishing: Get https://api-int.lbgsm9.lab.c-cor.com:6443/apis/storage.k8s.io/v1/csinodes/lbgsm9-hh4f4-master-0: dial tcp 10.109.180.9:6443: connect: connection refused
Jul 22 18:43:47 lbgsm9-hh4f4-master-0 hyperkube[1620]: E0722 18:43:47.897982    1620 kubelet.go:2285] node "lbgsm9-hh4f4-master-0" not found
Jul 22 18:43:47 lbgsm9-hh4f4-master-0 hyperkube[1620]: E0722 18:43:47.998392    1620 kubelet.go:2285] node "lbgsm9-hh4f4-master-0" not found
Jul 22 18:43:48 lbgsm9-hh4f4-master-0 hyperkube[1620]: I0722 18:43:48.017673    1620 kubelet.go:1989] SyncLoop (SYNC): 1 pods; recyler-pod-lbgsm9-hh4f4-master-0_openshift-infra(737d98e4436eec50342085eec891e230)
Jul 22 18:43:48 lbgsm9-hh4f4-master-0 hyperkube[1620]: I0722 18:43:48.017731    1620 kubelet.go:2034] Pod "recyler-pod-lbgsm9-hh4f4-master-0_openshift-infra(737d98e4436eec50342085eec891e230)" has completed, ignoring remaining sync work: sync

**Comment by lbrigman124** [comment]
Second capture.  Not sure if there is more.  Added the master IP to the capture command.
[log-bundle-20200721154136.tar.gz](https://github.com/openshift/okd/files/4962538/log-bundle-20200721154136.tar.gz)

Looking at the install log I find this info:
time="2020-07-22T13:18:34-07:00" level=debug msg="lost connection"
time="2020-07-22T13:18:34-07:00" level=debug msg="Warning: Permanently added '10.109.180.68' (ECDSA) to the list of known hosts.\r"
time="2020-07-22T13:18:34-07:00" level=debug msg="core@10.109.180.68: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).\r"

I can ssh into the master node why is it not getting there when collecting logs?

**Comment by vrutkovs** [comment]
>Jul 22 18:43:47 lbgsm9-hh4f4-master-0 hyperkube[1620]: E0722 18:43:47.998392 1620 kubelet.go:2285] node "lbgsm9-hh4f4-master-0" not found

it appears it can't contact the bootstrap node, very odd.

>time="2020-07-22T13:18:34-07:00" level=debug msg="core@10.109.180.68: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).\r"

Make sure you're using the same ssh key as specified in the install-config.

**Comment by jcpowermac** [comment]
Can we make sure that the hostname is being set?
The installer sets the extra config on the virtual machine `guestinfo.hostname` and then a service on RHCOS picks up that extra config string setting the hostname.

Also VMware Tools is critical, please make sure you see results within vCenter.

**Comment by lbrigman124** [comment]
I set the ssh key via the openshift-installer and use the same key via ssh-agent to login to the nodes.
Looks like the hostname is set correctly:
[core@lbgsm9-hh4f4-master-0 ~]$ hostname
lbgsm9-hh4f4-master-0
[core@lbgsm9-hh4f4-master-0 ~]$ hostname -f
lbgsm9-hh4f4-master-0.lbgsm9.lab.c-cor.com
[core@lbgsm9-hh4f4-master-0 ~]$ exit

$ govc vm.info lbgsm9-hh4f4-master-0
Name:           lbgsm9-hh4f4-master-0
  Path:         /xxxx/vm/lbgsm9-hh4f4/lbgsm9-hh4f4-master-0
  UUID:         420c2479-b322-b106-b5c3-7e8d4f4d7d67
  Guest name:   Red Hat Enterprise Linux 7 (64-bit)
  Memory:       16384MB
  CPU:          4 vCPU(s)
  Power state:  poweredOn
  Boot time:    2020-07-21 21:55:49.23596 +0000 UTC
  IP address:   10.109.180.68
  Host:         ucs-vmware-esxi-8.lab.c-cor.com

I also see the name in for both Vcenter though the bootstrap node hostname is localhost.

Both machines have this in the dmesg (console output):
[161816.455724] audit: type=1130 audit(1595530889.414:66303): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=zincati comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[161816.459623] audit: type=1131 audit(1595530889.414:66304): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=zincati comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[161816.463463] audit: type=1130 audit(1595530889.416:66305): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=zincati comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[161816.862239] audit: type=1131 audit(1595530889.821:66306): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=zincati comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
[161820.117032] audit: type=2404 audit(1595530893.075:66307): pid=1114299 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:sshd_t:s0-s0:c0.c1023 msg='op=destroy kind=server fp=SHA256:98:79:eb:a2:55:9b:84:fa:95:3d:92:89:42:c4:07:ce:9d:44:2a:b9:02:8b:ad:53:9f:96:f2:e2:a4:2c:59:68 direction=? spid=1114299 suid=0  exe="/usr/sbin/sshd" hostname=? addr=? terminal=? res=success'

**Comment by lbrigman124** [comment]
Both the bootstrap node and the master node have the same secondary  IP address.  The one that is registered as the cluster IP.

**Comment by lbrigman124** [comment]
If the existing logs don't point this out; this was a single all-in-one install without a front-end load balancer.
What other information do you want to see?

**Comment by abaxo** [comment]
This looks like the same behaviour that I was seeing with vmware 7.0 (I dont think version matters). Interestingly after running builds for OKD 4.5 and OpenShift 4.5 (including latest nightlies as of yesterday), neither platform would provide the Ignition configs to the masters - always times out. The same config (only difference between them being the platform config) works against ovirt 4.4 perfectly every time.

OKD 4.5 latest GA presents a server error presenting the ignition config, but Openshift 4.5 serves it. The result in both cases is that the ignition config grab on the master side times out.

Last night I tried builds of both Openshift and OKD 4.6 nightlies and they worked a treat, so hopefully there is a fix there somewhere that just needs backporting.

**Comment by vrutkovs** [comment]
https://github.com/openshift/machine-config-operator/pull/1900 includes a few fixes to stabilize hostname detection

**Comment by lbrigman124** [comment]
Figured out why the log collection is failing.  I'm not using a default key and the logging is grabbing my default key.
`
INFO Pulling debug logs from the bootstrap machine
DEBUG Added /tmp/bootstrap-ssh661808103 to installer's internal agent
DEBUG Added /home/lbrigman/.ssh/id_rsa to installer's internal agent
DEBUG Gathering bootstrap systemd summary ...
DEBUG Gathering bootstrap failed systemd unit status ...
DEBUG Gathering bootstrap journals ...
`
The key is not the one I have in the install-config.yaml so it isn't in the master node.

**Comment by lbrigman124** [comment]
Logs from the 4.6 build from 8/1 that failed but I have the master node logs because of the ssh key change that I noted above.
[log-bundle-20200807192120.tar.gz](https://github.com/openshift/okd/files/5053251/log-bundle-20200807192120.tar.gz)

**Comment by lbrigman124** [comment]
Nothing from the logs or other feedback provided any help.
Today, I tried to install using the latest stable release: 4.5.0-0.okd-2020-09-04-180756
It told me that I couldn't have both the API and the VIP on the same IP.  
I changed that in the install-config and the install completed successfully.
Nothing in the previous versions said that it wasn't acceptable to do that.
Considering this was a single node install, that wasn't obvious from the docs nor messages from the previous versions.
